lib/engine/game/g_1862/game.rb

```diff
@@ -1365,6 +1427,57 @@ def revenue_str(route)
         def compute_other_paths(_routes, _route)
           []
         end
+
+        def train_limit(entity)
+          @phase.train_limit(entity) * (@phase.available?('G') ? 1 : 3)
+        end
+
+        def train_limit_by_type(entity)
+          @phase.train_limit(entity)
+        end
+
+        def used_train_price(train)
+          train.sym == @phase.name ? train.price : (train.price / 2).to_i
+        end
+
+        def info_train_name(train)
+          train.names_to_prices.keys.find { |n| /^[A-H]$/ =~ n } + ': ' +
```

Comment:
what is this stuff

Reply:
There are 8 train levels, A-H, that correspond to each of the phases. Within each level, there are 3 train types that can be bought (freight, local and express). I started out with this "stuff" but the info page was confusing. The first C train, for instance will rust all A trains, but the train/phase roster only showed that a 3F would rust a 1F. This method will show the train level followed by the trains at that level.

In addition, I collapse all of the prices for a single level down to one, since they are all the same.

Reply:
Removed regexps